lib/sorbet-rails/rails_mixins/pluck_to_tstruct.rb

```diff
@@ -31,13 +31,14 @@ def pluck_to_tstruct(ta_struct, associations: {})
     end
 
     pluck_keys = (tstruct_keys - associations_keys) + associations.values
+    tstruct_keys_in_pluck_order = tstruct_keys - associations_keys + associations_keys
 
     # loosely based on pluck_to_hash gem
     # https://github.com/girishso/pluck_to_hash/blob/master/lib/pluck_to_hash.rb
     keys_one = pluck_keys.size == 1
     pluck(*pluck_keys).map do |row|
       row = [row] if keys_one
-      value = Hash[map_nil_values_to_default(tstruct_props, tstruct_keys.zip(row))]
+      value = Hash[map_nil_values_to_default(tstruct_props, tstruct_keys_in_pluck_order.zip(row))]
```

Comment:
When we zip the tstruct keys with the plucked values, we want to make sure the corresponding key and value match up correctly.

This requires that our tstruct keys are in the same order as the `pluck_keys`, which is why we want to subtract the `associations_keys` and then append them to the end of `tstruct_keys` on line 34.